pandas/core/groupby/ops.py

```diff
@@ -655,15 +655,15 @@ def agg_series(self, obj, func):
             return self._aggregate_series_fast(obj, func)
         except AssertionError:
             raise
-        except ValueError as err:
+        except (ValueError, TypeError) as err:
             if "No result." in str(err):
                 # raised in libreduction
                 pass
             elif "Function does not reduce" in str(err):
                 # raised in libreduction
```

Comment:
@jbrockmendel is it the intention to keep this long term, or is this also planned to be cleaned-up in a follow-up?

Reply:
cleaned

Reply:
And what's the idea of how to clean this up? 
(we could also raise a more specific internal error (subclass) to catch)

Reply:
> (we could also raise a more specific internal error (subclass) to catch)

That's what I'm leaning towards, yah